src/lib/int.ts

```diff
@@ -202,6 +203,38 @@ class UInt64 extends CircuitValue {
     return new UInt64(z);
   }
 
+  /**
+   * Bitwise XOR gate on {@link UInt64} elements. Equivalent to the [bitwise XOR `^` operator in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_XOR).
+   * A XOR gate works by comparing two bits and returning `1` if two bits differ, and `0` if two bits are equal. It applies XOR to all 64 bits of the elements.
+   * ```typescript
+   * let a = UInt64.from(5);    // ... 000101
+   * let b = UInt64.from(3);    // ... 000011
+   *
+   * let c = a.xor(b);    // ... 000110
+   * c.assertEquals(6);
+   * ```
+   */
+  xor(y: UInt64) {
+    return new UInt64(this.value.xor(y.value, UInt64.NUM_BITS));
+  }
+
+  /**
+   * A (left and right) rotation is similar to the shift operation, `<<` and `>>` in JavaScript, just that bits are being appended to the other side.
+   * `direction` is a boolean, defining the direction of the rotation - `left - true` and `right = false`
+   *
+   * ```typescript
+   * let a = UInt64.from(12);
+   * let b = a.rot(2, true);  // left rotation by 2 bit
+   * c.assertEquals(20);
+   * ```
+   *
+   * @param bits amount of bits to rotate this {@link UInt64} element with.
+   * @param direction (true) left or (false) right rotation direction.
+   */
+  rot64(bits: number, direction: boolean = true) {
```

Comment:
nit: I would call this just `rot`, not `rot64`, since the 64 bit length is already implicit in using a `UInt64`